subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -490,10 +492,32 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 			}
 			break;
 		case 'k':
-			params->security = atoi(optarg);
+			params->security = BIT(WIFI_SECURITY_TYPE_PSK);
```

Comment:
If the user gives "none", still PSK bit will be set(as it's being set default). Please check.